lib/express_validator.js

```diff
@@ -340,5 +339,47 @@ function locate(req, name) {
   return undefined;
 }
 
+/**
+ * format param output if passed in as array (for nested)
+ * before calling errorFormatter
+ *
+ * @method param
+ * @param  {(string|string[])} param       parameter as a string or array
+ * @param  {string} msg
+ * @param  {string} value
+ * @return {function}
+ */
+function formatErrors(param, msg, value) {
+  var formattedParam = formatParamOutput(param);
+
+  return this.errorFormatter(formattedParam, msg, value);
+}
+
+// Convert nested params as array into string for output
+// Ex: ['users', '0', 'fields', 'email'] to 'users[0].fields.email'
+function formatParamOutput(param) {
+  if (Array.isArray(param)) {
+    param = param.reduce(function(prev, curr) {
```

Comment:
That regex only replaced the first number it found (no global flag) and it didn't replace it if the number it found was at the end of the string (since it wouldn't match `/\.([0-9]+)\./`). I added a few tests to account for these scenarios in case you wanted to try tweaking your regex suggestion.